server/server.go

```diff
@@ -943,3 +946,57 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startReloadWatcherForMirrorRegistries starts a new goroutine that
+// monitors the "registries.conf.d" directory. If not set, it defaults to
+// "/etc/containers/registries.conf.d". The routine reloads the registries
+// configuration when a file is modified or created in the directory.
+func (s *Server) startReloadWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, using default")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
```

Comment:
A candidate to make into a constant.

Reply:
What about the default `/etc/containers/registries.conf` file?

Tere is also the `XDG_CONFIG_HOME` (albeit, I am not sure if this is strictly followed by the underlying package) where the user can put their configuration into `${HOME}/.config/containers/registries.conf`.

Reply:
There might be issues watching a single file, though.

Reply:
I'm unsure about scenarios where users directly modify /etc/containers/registries.conf. I thought we should monitor user-configured drop-in directories or `/etc/containers/registries.conf.d`. Currently, Cri-O reloads the configuration with a `SIGHUP` signal. Given the mirror registries reloading issue, it seems unlikely users would directly alter `registries.conf`, but this is just my personal opinion. To comprehensively address use cases, we should consider edge cases as well.

Reply:
I think there are functions in c/storage somewhere to get this file correctly. I think we probably should use them in case a user uses crio as a non root user (though this is not really supported). I also think we can ignore registires.conf for this feature, and focus on conf.d. That seems like a reasonable limitation and simplifies the code

Reply:
I can imagine users will directly modify

Reply:
I had an offline discussion with @mtrmac regarding the deprecation note mentioned above. It seems that the function will reliably trigger a reload without a problem. The deprecation note is specifically related to the return value, now incorporating more configuration items than the utilized V2RegistriesConf type. Also, we don't use the configuration returned by that function anywhere in the codebase.

Reply:
the original point still stands: is this the best way to find the registries directory?

Reply:
The c/image answer is “don’t”.

There _is_ `sysregistriesv2.ConfigDirPath`, but that’s deprecated because we added more places where configuration can be found since, and c/image doesn’t want to constrain itself for the future not to ever add any _more_ places; so a stable API like `ConfigDirPath` promising “this is where data will be read from” is making unsustainable commitments about the future.

So we now have `ConfigurationSourceDescription` but that’s _explicitly_ designed to not be machine-consumable. (Some of the sources might not files at all, in principle…)

---

I realize that’s not all that helpful.

The supported path is to invert the flow, for the caller to _define_ the location (via `SystemContext.SystemRegistriesConfDirPath`) and for c/image to respect that choice. But that’s, of course, not the same thing, and basically not answering the “so which path should be used” question at all.

Reply:
Thank you, @mtrmac, for the details! Appreciated.